### PR TITLE
PV2-538 required payments to handle payable earning events

### PR DESCRIPTION
**Pull request by kuzinator**

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.Application/Interfaces/IEarningEventHandler.cs [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
 using System.Threading.Tasks;
+using SFA.DAS.Payments.DataLocks.Messages;
 using SFA.DAS.Payments.EarningEvents.Messages.Events;
+using SFA.DAS.Payments.Messages.Core.Events;
 using SFA.DAS.Payments.RequiredPayments.Messages.Events;
 using SFA.DAS.Payments.RequiredPayments.Model.Entities;
 
 namespace SFA.DAS.Payments.RequiredPayments.Application
 {
-    public interface IEarningEventHandler
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.Application/Interfaces/IEarningEventHandler.cs]: filename is nothing like the 4 interfaces in this file

File: src/SFA.DAS.Payments.RequiredPayments.Application/Processors/ApprenticeshipContractTypeEarningEventProcessor.cs [hunk 2]
```diff
@@ -13,40 +14,43 @@
 
 namespace SFA.DAS.Payments.RequiredPayments.Application.Processors
 {
-    public class ApprenticeshipContractType2EarningEventProcessor : EarningEventProcessorBase<ApprenticeshipContractType2EarningEvent, RequiredPaymentEvent>
+    public class ApprenticeshipContractType2EarningEventProcessor : ApprenticeshipContractTypeEarningEventProcessor<ApprenticeshipContractType2RequiredPaymentEvent, ApprenticeshipContractType2EarningEvent>, IApprenticeshipContractType2EarningsEventProcessor
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.Application/Processors/ApprenticeshipContractTypeEarningEventProcessor.cs]: too many classes in this file

File: src/SFA.DAS.Payments.Audit.Application.UnitTests/SFA.DAS.Payments.Audit.Application.UnitTests.csproj [hunk 3]
```diff
@@ -13,9 +13,9 @@
     <PackageReference Include="Autofac" Version="4.8.1" />
     <PackageReference Include="AutoMapper" Version="7.0.1" />
     <PackageReference Include="AutoMoqCore" Version="1.0.0" />
-    <PackageReference Include="FluentAssertions" Version="5.5.3" />
+    <PackageReference Include="FluentAssertions" Version="5.4.2" />
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.Audit.Application.UnitTests/SFA.DAS.Payments.Audit.Application.UnitTests.csproj]: yes, this was a consolidation excercise to get rid of warnings and deployment issues, I took most used version as a basis for consolidation

File: src/SFA.DAS.Payments.Audit.Application.UnitTests/SFA.DAS.Payments.Audit.Application.UnitTests.csproj [hunk 4]
```diff
@@ -13,9 +13,9 @@
...
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
-    <PackageReference Include="MSTest.TestAdapter" Version="1.4.0" />
+    <PackageReference Include="MSTest.TestAdapter" Version="1.3.2" />
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.Audit.Application.UnitTests/SFA.DAS.Payments.Audit.Application.UnitTests.csproj]: yes, this was a consolidation excercise to get rid of warnings and deployment issues, I took most used version as a basis for consolidation

File: src/SFA.DAS.Payments.DataLocks.DataLockService.Interfaces/IDatalocksService.cs [hunk 5]
```diff
@@ -11,6 +11,6 @@ namespace SFA.DAS.Payments.DataLocks.DataLockService.Interfaces
 {
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.DataLocks.DataLockService.Interfaces/IDatalocksService.cs]: They should, I missed this one. Fixed.

File: src/SFA.DAS.Payments.EarningEvents.AcceptanceTests/app.config [hunk 6]
```diff
@@ -92,6 +92,10 @@
         <assemblyIdentity name="System.Runtime" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-4.1.2.0" newVersion="4.1.2.0" />
       </dependentAssembly>
+      <dependentAssembly>
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.EarningEvents.AcceptanceTests/app.config]: It needs to be removed. Done.

File: src/SFA.DAS.Payments.Monitoring.AcceptanceTests/SFA.DAS.Payments.Monitoring.AcceptanceTests.csproj [hunk 7]
```diff
@@ -126,11 +136,11 @@
     <Reference Include="System.Data.SqlClient, Version=4.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\packages\System.Data.SqlClient.4.6.0\lib\net461\System.Data.SqlClient.dll</HintPath>
     </Reference>
-    <Reference Include="System.Diagnostics.DiagnosticSource, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
-      <HintPath>..\packages\System.Diagnostics.DiagnosticSource.4.5.0\lib\net46\System.Diagnostics.DiagnosticSource.dll</HintPath>
+    <Reference Include="System.Diagnostics.DiagnosticSource, Version=4.0.3.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Diagnostics.DiagnosticSource.4.5.1\lib\net46\System.Diagnostics.DiagnosticSource.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.Monitoring.AcceptanceTests/SFA.DAS.Payments.Monitoring.AcceptanceTests.csproj]: 4.5.1 is the correct package version

File: src/SFA.DAS.Payments.PerformanceTests/SFA.DAS.Payments.PerformanceTests.csproj [hunk 8]
```diff
@@ -203,8 +207,8 @@
       <HintPath>..\packages\System.Runtime.CompilerServices.Unsafe.4.5.1\lib\netstandard2.0\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
     </Reference>
     <Reference Include="System.Runtime.Serialization" />
-    <Reference Include="System.Runtime.Serialization.Primitives, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\packages\System.Runtime.Serialization.Primitives.4.1.1\lib\net46\System.Runtime.Serialization.Primitives.dll</HintPath>
+    <Reference Include="System.Runtime.Serialization.Primitives, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Runtime.Serialization.Primitives.4.3.0\lib\net46\System.Runtime.Serialization.Primitives.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.PerformanceTests/SFA.DAS.Payments.PerformanceTests.csproj]: yes, that's correct

File: src/SFA.DAS.Payments.RequiredPayments.Application/Interfaces/IEarningEventHandler.cs (hunk 1 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.Application/Interfaces/IEarningEventHandler.cs]: Renamed

File: src/SFA.DAS.Payments.RequiredPayments.Application/Processors/ApprenticeshipContractTypeEarningEventProcessor.cs (hunk 2 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.Application/Processors/ApprenticeshipContractTypeEarningEventProcessor.cs]: I've split them now.

File: src/SFA.DAS.Payments.Messages.Core/Events/EarningEvent.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿//using System.Collections.ObjectModel;
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.Messages.Core/Events/EarningEvent.cs]: removed

File: src/SFA.DAS.Payments.Model.Core/CollectionPeriod.cs [hunk 10]
```diff
@@ -6,7 +6,11 @@ namespace SFA.DAS.Payments.Model.Core
     public class CollectionPeriod 
     {
         public short AcademicYear { get; set; }
-        //public string Name => $"{AcademicYear}-R{Period:00}";  //TODO: Move this property out of this class.  Only tests use this property so should not pollute the services.
+        public string Name => $"{AcademicYear}-R{Period:00}";  //TODO: Move this property out of this class.  Only tests use this property so should not pollute the services.
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.Model.Core/CollectionPeriod.cs]: moved to extension method in tests project

**Pull request merged**